src/main/java/io/jenkins/pluginhealth/scoring/probes/ProbeEngine.java

```diff
@@ -47,19 +50,23 @@ public class ProbeEngine {
 
     private final List<Probe> probes;
     private final PluginService pluginService;
+    private final UpdateCenterService updateCenterService;
 
-    public ProbeEngine(List<Probe> probes, PluginService pluginService) {
+    public ProbeEngine(List<Probe> probes, PluginService pluginService, UpdateCenterService updateCenterService) {
         this.probes = List.copyOf(probes);
         this.pluginService = pluginService;
+        this.updateCenterService = updateCenterService;
     }
 
     /**
      * Starts to apply all the {@link Probe} implementations on all the plugins registered in the database.
      */
-    public void run() {
+    public void run() throws IOException {
         LOGGER.info("Start running probes on all plugins");
+        final UpdateCenter updateCenter = updateCenterService.fetchUpdateCenter();
         pluginService.streamAll().parallel()
             .peek(plugin -> {
+                final ProbeContext probeContext = ProbeContext.withUpdateCenter(updateCenter);
```

Comment:
Optional question, answer only if you have time (please feel free to ignore):

1. Why is the `withUpdateCenter()` static?
What if it wasn't static and I called it from here like this: `final ProbeContext probeContext = new ProbeContext(updateCenter);`? Did you make it static **only** to avoid the `May expose internal representation ...` error?

2. Why does each plugin gets a new separate instance of the `probeContext` object?
As per my understanding, all the update-center-related probes are just using this context to access the update center's representation. Wouldn't this still work the same way if we just instantiate this object once?

Reply:
> 1.
Because I found it easier at the time.

> 2. 
Because at some point, the context will hold more that just the update-center representation but intermediate state of the probes executions to carry details from probe to probe without putting that into a `ProbeResult`